pandas/core/generic.py

```diff
@@ -8564,6 +8523,246 @@ def nanptp(values, axis=0, skipna=True):
+    Note that if the function modifies the index it must do so consistently
+    in the case of a DataFrame.
+
+    See also
+    --------
+    pandas.DataFrame.aggregate : Perform aggregate type of operations.
+    pandas.DataFrame.apply : Apply general functions.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame(np.arange(30).reshape(10, 3),
+    ...                   columns=['A', 'B', 'C'],
+    ...                   index=pd.date_range('1/1/2000', periods=10))
+    >>> df.iloc[3:7] = np.nan
+    >>> df
+                A     B     C
+    2000-01-01   0.0   1.0   2.0
+    2000-01-02   3.0   4.0   5.0
+    2000-01-03   6.0   7.0   8.0
+    2000-01-04   NaN   NaN   NaN
+    2000-01-05   NaN   NaN   NaN
+    2000-01-06   NaN   NaN   NaN
+    2000-01-07   NaN   NaN   NaN
+    2000-01-08  21.0  22.0  23.0
+    2000-01-09  24.0  25.0  26.0
+    2000-01-10  27.0  28.0  29.0
+
+    By default, `func` is applied to each column independently.
+
+    >>> df.transform(lambda x: (x - x.mean()) / x.std())
+                A         B         C
+    2000-01-01 -1.143001 -1.143001 -1.143001
+    2000-01-02 -0.889001 -0.889001 -0.889001
+    2000-01-03 -0.635001 -0.635001 -0.635001
+    2000-01-04       NaN       NaN       NaN
+    2000-01-05       NaN       NaN       NaN
+    2000-01-06       NaN       NaN       NaN
+    2000-01-07       NaN       NaN       NaN
+    2000-01-08  0.635001  0.635001  0.635001
+    2000-01-09  0.889001  0.889001  0.889001
+    2000-01-10  1.143001  1.143001  1.143001
+
+    Apply `func` to each row with ``axis='columns'``.
+
+    >>> df.transform(lambda x: (x - x.mean()) / x.std(), axis='columns')
+                A    B    C
+    2000-01-01 -1.0  0.0  1.0
+    2000-01-02 -1.0  0.0  1.0
+    2000-01-03 -1.0  0.0  1.0
+    2000-01-04  NaN  NaN  NaN
+    2000-01-05  NaN  NaN  NaN
+    2000-01-06  NaN  NaN  NaN
+    2000-01-07  NaN  NaN  NaN
+    2000-01-08 -1.0  0.0  1.0
+    2000-01-09 -1.0  0.0  1.0
+    2000-01-10 -1.0  0.0  1.0
+
+    >>> def f(s):
+    ...     "input is a Series"
+    ...     i = np.arange(s.size)
+    ...     return s + i
+
+    >>> df.transform(f)
+                A     B     C
+    2000-01-01   0.0   1.0   2.0
+    2000-01-02   4.0   5.0   6.0
+    2000-01-03   8.0   9.0  10.0
+    2000-01-04   NaN   NaN   NaN
+    2000-01-05   NaN   NaN   NaN
+    2000-01-06   NaN   NaN   NaN
+    2000-01-07   NaN   NaN   NaN
+    2000-01-08  28.0  29.0  30.0
+    2000-01-09  32.0  33.0  34.0
+    2000-01-10  36.0  37.0  38.0
+
+    >>> df.transform(f, axis=1)
+                A     B     C
+    2000-01-01   0.0   2.0   4.0
+    2000-01-02   3.0   5.0   7.0
+    2000-01-03   6.0   8.0  10.0
+    2000-01-04   NaN   NaN   NaN
+    2000-01-05   NaN   NaN   NaN
+    2000-01-06   NaN   NaN   NaN
+    2000-01-07   NaN   NaN   NaN
+    2000-01-08  21.0  23.0  25.0
+    2000-01-09  24.0  26.0  28.0
+    2000-01-10  27.0  29.0  31.0
+
+    >>> def g(s):
+    ...     "input is a Series whose index may be modified"
```

Comment:
I wonder whether this is officially supported, or whether it just happens to work... I'm struggling to find a reason why this would be the best way to do things.

Reply:
Just expanding on the fact that the input is a Pandas.Series: All relevant manipulation works.  
I did not try and think if that was purposefully intended or is a consequence of implementation.  
Just to help a user, like me, see in a glance what you can do.

Reply:
>  Just to help a user, like me, see in a glance what you can do.

What you can do *currently* :) I could imagine a future change breaking this.

If you're interested, you could go through the NDFrame.transform tests to see if this is tested. If it's tested then I'd say it's supported, otherwise, I'd say it's not.

Reply:
For now I removed the index modifying example.